binary_search_tree/tree.py

```diff
@@ -14,21 +14,106 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n) if the tree is balanced
+    # Space Complexity: O(1) - we are just creating one object (the new node)
```

Comment:
🪐 The space complexity here will actually be O(log n) as well if you implement this recursively s iy looks like you begin to in `add_helper` because of the recursive call stack, but you are correct it will be O(1) for the iterative solution you implemented